ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -7822,9 +7824,18 @@ protected Operator genFileSinkPlan(String dest, QB qb, Operator input)
 
     List<ColumnInfo> vecCol = new ArrayList<ColumnInfo>();
 
-    if (updating(dest) || deleting(dest)) {
+    boolean nonNativeAcid = Optional.ofNullable(destinationTable)
+        .map(Table::getStorageHandler)
+        .map(HiveStorageHandler::supportsAcidOperations)
+        .orElse(false);
+    boolean isUpdateDelete = updating(dest) || deleting(dest);
+    if (!nonNativeAcid && isUpdateDelete) {
```

Comment:
Is it a valid situation that: isUpdateDelete and we need to go to the `else`?
If not it might be easier to read:
```
if (updating(dest) || deleting(dest)) {
  if (nonNativeAcid) {
...
  } else {
...
} else {
..
}
```

Reply:
I agree, that's more readable